src/com/geekhub/hw2/cars/SolarCar.java

```diff
@@ -0,0 +1,24 @@
+package com.geekhub.hw2.cars;
+
+import com.geekhub.hw2.parts.interfaces.EnergySource;
+import com.geekhub.hw2.parts.interfaces.Engine;
+import com.geekhub.hw2.parts.interfaces.Gear;
+import com.geekhub.hw2.parts.interfaces.RunMechanism;
+
+public class SolarCar extends Car {
+
+    public SolarCar(EnergySource energySource, Engine engine, Gear gear, RunMechanism runMechanism) {
```

Comment:
Yes, I agree with this. But I have problem while I am trying to fix it. SolarCar extends Car, Car implements Vehicle. So, when I passed concrete implementation:
  public SolarCar(Accumulator accumulator, SolarCarEngine engine, CarTransmission transmission, Wheels wheels) {
        super(accumulator, engine, transmission, wheels);
    }
and
public Car(GasTank gasTank, CarEngine engine, CarTransmission transmission, Wheels wheels) {
        super(gasTank, engine, transmission, wheels);
    }
I got an error (it is obviosly), that types can not be converted. How I can resolve it?